test-clients/src/main/java/com/hedera/services/bdd/spec/assertions/ContractFnResultAsserts.java

```diff
@@ -271,4 +276,27 @@ private static Optional<Throwable> matchErrors(Object[] expecteds, Object[] actu
 		}
 		return Optional.empty();
 	}
+
+	private static Optional<Throwable> validateRandomResult(final Object[] expecteds, final Object[] actuals) {
+		try {
+			for (int i = 0; i < Math.max(expecteds.length, actuals.length); i++) {
+				Object expected = expecteds[i];
+				Object actual = actuals[i];
+				Assertions.assertNotNull(expected);
+				Assertions.assertNotNull(actual);
+				if (expected instanceof byte[]) {
+					Assertions.assertEquals(((byte[]) expected).length, ((byte[]) actual).length);
+				} else if (expected instanceof Integer) {
+					Assertions.assertTrue(((BigInteger) actual).intValue() >= 0 &&
+							((BigInteger) actual).intValue() < ((Integer) expected).intValue());
```

Comment:
Q: what's the logic being relied on and tested here?
Seems like the `intValue` of the `BigInteger` is always smaller than the `intValue` of the `Integer`.
Is this the precision of the two types here before int conversion?

Reply:
I am checking if the actual value (pseudo random number generated) is less than expected value(provided range). Ad mentioned in the HIP , the pseudo random number should be generated with `0 <= X < range`